src/components/DotIndicatorMessage.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import {StyleProp, TextStyle, View, ViewStyle} from 'react-native';
+import * as Localize from '@libs/Localize';
+import styles from '@styles/styles';
+import themeColors from '@styles/themes/default';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import Text from './Text';
+
+type DotIndicatorMessageProps = {
+    messages: Record<string, string>;
```

Comment:
The propType says that this should be the structure of the type:

```ts
    messages: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.object]))])),
```

And we have evidence that we can indeed pass an array to the record, [see here](https://github.com/Expensify/App/blob/main/src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.js#L283).

I believe the structure would be something like this:

```ts
    messages: Record<string, string | [string, Array<Record<string, unknown>>]>,
```

Reply:
Yep, you are right, let's hold this PR for now. Until [Localize](https://github.com/Expensify/App/pull/29742) is migrated. I think we need to adjust type for `translateIfPhraseKey` 🤔

Reply:
@fabioh8010 Here's my [idea](https://github.com/Expensify/App/pull/29742/files/c352b5339525e485ea0db2ccbf49ba4ea073e785..1a569e420b59f3a50de924689a14f38ca0298be9). I made the function very generic as I wasn't able to type it with more safety, keep in mind how it is used, for example in this component. Any ideas how to improve it?

Reply:
I think it's okay, because the way how `translateIfPhraseKey` is being used it would cause issues on other components because we just pass a generic `error` to it.